care/facility/api/viewsets/patient.py

```diff
@@ -609,9 +647,19 @@ class FacilityDischargedPatientViewSet(GenericViewSet, mixins.ListModelMixin):
         rest_framework_filters.OrderingFilter,
         PatientCustomOrderingFilter,
     )
-    filterset_class = PatientFilterSet
+    filterset_class = DischargePatientFilterSet
     queryset = (
-        PatientRegistration.objects.select_related(
+        PatientRegistration.objects.annotate(
```

Comment:
This might get expensive, if its really important can we just store it like last consultation?

Reply:
sure

Reply:
Although, if I make a field called `last_discharge_consultation` it would not work out... lets say a patient was discharged from two different facilities, then they would require different consultations in the discharge view set.

Reply:
>lets say a patient was discharged from two different facilities, then they would require different consultations in the discharge view set.

in that case, we should also filter the consultation subquery based on users accessible facility

Reply:
Utilizing `PatientDRYFilter` for this, that works? I think it should cover.